satpy/writers/cf_writer.py

```diff
@@ -259,22 +260,14 @@ def area2cf(dataarray, strict=False):
     return res
 
 
-def make_time_bounds(dataarray, start_times, end_times):
+def make_time_bounds(start_times, end_times):
     """Create time bounds for the current *dataarray*."""
-    import numpy as np
     start_time = min(start_time for start_time in start_times
                      if start_time is not None)
     end_time = min(end_time for end_time in end_times
                    if end_time is not None)
-    try:
-        dtnp64 = dataarray['time'].data[0]
-    except IndexError:
-        dtnp64 = dataarray['time'].data
-    time_bnds = [(np.datetime64(start_time) - dtnp64),
-                 (np.datetime64(end_time) - dtnp64)]
-    data = xr.DataArray(np.array(time_bnds)[None, :] / np.timedelta64(1, 's'),
+    data = xr.DataArray([[np.datetime64(start_time), np.datetime64(end_time)]],
```

Comment:
I'm not entirely sure, why the current method uses the timestamp. But I guess it's just for encoding reasons (bounds relative to timestamp). By changing the bounds' datatype to `datetime64` xarray will handle the encoding.